core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -218,6 +204,7 @@ public DynamicResourcePool(Options options) {
 
     mMaxCapacity = options.getMaxCapacity();
     mMinCapacity = options.getMinCapacity();
+    mResourceAvailable = new ArrayDeque<>(Math.min(mMaxCapacity, 32));
```

Comment:
Because this class is mostly used for connection pool, i assume 32 is a reasonable starting point for the connection pool size. I didn't make it a constant because we don't seem to be doing that in similar places like (ConcurrentHashSet). I am happy to make it a constant if you prefer.